lib/vagrant-libvirt/action/handle_box_image.rb

```diff
@@ -102,13 +102,15 @@ def call(env)
             # Upload box image to storage pool
             ret = upload_image(box_image_file, config.storage_pool_name,
                                env[:box_volume_name], env) do |progress|
-              env[:ui].clear_line
-              env[:ui].report_progress(progress, box_image_size, false)
+              env[:ui].rewriting do |ui|
```

Comment:
It appears this is quite a new API, so if we add this here, the plugin will break for all previous versions of vagrant. While this isn't an issue for some users, there have been enough issues with trying to use the upstream release of Vagrant on newer fedora releases due to mismatches in library versions that it would be better to try and avoid breaking compatibility

Maybe add a utility function that can yield based on whether `env[:ui]` responds to the rewriting method or not.

Just knocking this out from memory rather than checking that it works:
```
def rewriting(ui) do
    if ui.responds_to?(:rewriting)
        yield ui.rewriting
    else
        yield ui
    end
end
```

Then if we call this utility with `env[ui]` we should get the new behaviour for newer versions of vagrant while continuing to work with older releases, this code here would become:

```
    util.rewriting(env[:ui]) do |ui|
        ui.clear_line
        ui.report_progress(progress, box_image_size, false)
    end
```

Should be able to add the helper method under the util directory https://github.com/vagrant-libvirt/vagrant-libvirt/tree/master/lib/vagrant-libvirt/util, and simply include the module as needed.

Reply:
Thank you, that definitely makes sense. I had to tweak the `rewriting()` method a bit, otherwise it would throw exceptions as the `Vagrant::UI::Inteface::rewriting()` method yields `self` or nothing [depending on the object](https://github.com/hashicorp/vagrant/blob/master/lib/vagrant/ui.rb).

However, as this is my first experience with Ruby, I'm not sure if mine approach is the best, so please feel free to kick me in the right direction.